packages/front-end/components/Experiment/AnalysisSettingsBar.tsx

```diff
@@ -161,8 +165,12 @@ export default function AnalysisSettingsBar({
           ) : null}
           <div className="col-auto form-inline pr-5">
             <DimensionChooser
-              value={dimension}
+              value={precomputedDimension ?? dimension ?? ""}
               setValue={setDimension}
```

Comment:
suggestion: Would be nice to find a way to abstract this so we only deal with dimension and not 2 different separate fields for precomputed and ondemand.

Especially weird IMO where the `value` and the `setValue` do not match -- means we have some kind of implicit behavior somewhere that is likely harded to reason about.

Reply:
As per our discussion, you can see I'm using the `setAnalysisBarSettings`, but I still have to manage the state of the actual dimension and analysis settings using other methods for now. We could continue to refactor this if need be.